frontend/Plugin.js

```diff
@@ -0,0 +1,147 @@
+
+/**
+ * Tracking plugin to handle internal event tracking to the shopgate analytics sdk.
+ */
+class ShopgateAnalytics extends SgTrackingPlugin {
+  /**
+   * [constructor description]
+   * @param {Object} options The options passed to the plugin.
+   * @param {string} options.stage The stage / environment the plugin is running.
+   * @param {number} options.shopNumber The shop number.
+   * @param {number} [options.userId] The internal user id.
+   * @param {string} [options.pushToken] The push token of the device.
+   */
+  constructor({ stage, shopNumber, userId, pushToken }) {
+    super('sgAnalytics', { stage });
+
+    /* eslint-disable eslint-comments/no-unlimited-disable */
+    /* eslint-disable */
+    (function (src) { var s, r, t; r = false; s = document.createElement('script'); s.type = 'text/javascript'; s.src = src; t = document.getElementsByTagName('script')[0]; t.parentNode.insertBefore(s, t); window.__shopgate_aq = window.__shopgate_aq || []; window.sgAnalytics = function () { window.__shopgate_aq.push(arguments); }})
+    (`path/to/tracking-sdk.${stage}.min.js`);
+    /* eslint-enable */
+    /* eslint-enable eslint-comments/no-unlimited-disable */
+
+    sgAnalytics('setConfig', {
+      shopNumber,
+      pushToken,
+      sgUserId: userId,
+      channel: 'app',
+    });
+
+    this.registerEvents();
+  }
+
+  /**
+   * Registers all event handlers.
+   */
+  registerEvents() {
+    this.register.pageview((data, rawData) => {
+      let sdkData = {
+        url: data.page.merchantUrl,
+        type: rawData.page.name,
+      };
+
+      if (sdkData.type === 'productDetails' && rawData.product) {
+        sdkData = {
+          ...sdkData,
+          id: rawData.product.uid,
+          product: {
+            number: rawData.product.uid || rawData.product.productNumber,
+            name: rawData.product.name,
+            currency: rawData.product.amount.currency,
+            price: rawData.product.amount.displayPrice ||
+             rawData.product.amount.grosss ||
+             rawData.product.amount.net,
+            priceStriked: rawData.product.amount.displayPriceStriked ||
+              rawData.product.amount.gross ||
+              rawData.product.amount.net,
+            identifiers: {
+              ean: rawData.product.identifiers.ean,
+            },
+          },
+        };
+      }
+
+      sgAnalytics('track', 'pageViewed', sdkData);
+      console.error('pageview', data, rawData, sdkData);
+    });
+
+    this.register.addToCart((data, rawData) => {
+      console.log(rawData);
+      const sdkData = {
+        products: rawData.products.map(product => ({
+          number: product.uid || product.productNumber,
+          name: product.name,
+          currency: product.amount.currency,
+          price: product.amount.displayPrice ||
+           product.amount.grosss ||
+           product.amount.net,
+          priceStriked: product.amount.displayPriceStriked ||
```

Comment:
let's talk about the default value of priceStriked

Reply:
as discussed